design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,206 @@
+  - An text input field is available within the dropdown to filter the records based on partial or complete value matches. The list narrows down the records as the user types in a value. The system can search only the first five fields. Search across all types is supported.
+  - If there's a single match for the entered value, it should become highlighted, allowing the user to confirm the selection.
+  - If there aren't any matches, a suggestion to change the 'Search Columns' table preference is displayed.
+
+## Components
+
+### Record Selector
+
+The record selector component is used to retrieve records from other tables and add them as values to cells. The list of records is shown according to the referenced table set in a foreign key constraint column.
+
+### Table Selector
+
+The table selector allows users to select tables from a schema. This spec doesn't go into details on how this selector would work. It also doesn't consider the scenario for selecting both tables and views.
+
+### Record Preview
+
+The record preview allows users to identify records linked in other tables. It shows the first five field names and values. A specific limit for the value length needs to be defined so that space usage is optimal.
+
+### Table Constraints
+
+The table constraints settings list and provide details for all table-level constraints. Its most basic implementation will allow users to know which constraints exist and see the columns to which they apply. Users will also add or delete constraints but not edit them.
+
+### Table Preferences
+
+Table preferences will contain a group of options that users can set up to change specific properties of tables relevant only in the context of Mathesar, meaning they don't alter any database table properties. An example of this could be setting up a table's default search columns.
+
+## Other Considerations
+
+### Linked Records for Multi-Column Foreign Key Constraints
+
+- The current UI does not consider multi-column foreign key constraints and how those would be retrieved and selected. Record search and selection via the dropdown component will not be supported. Instead, the cells will be limited to input and edit text values. The text values will have to match the primary keys of the referenced records and be unique.
+
+### Usage of Color
+
+- A suggestion to use color as a means to differentiate table references has been included in this spec. However, implementation details need to be discussed before deciding if the user interface will use color elements to create a visual connection between elements.
+
+### Terminology
+
+- `Remove` and `Delete` are used throughout this spec and should be applied consistently across the entire user interface as they are different affordances. `Remove` will be used for data that can be added back, while `Delete` will indicate data destruction.
```

Comment:
We actually decided to use Delete for everything in https://github.com/centerofci/mathesar/discussions/872